tests/unit/completion/test_completer.py

```diff
@@ -299,6 +303,12 @@ def test_quickcomplete_flicker(status_command_stub, completer_obj,
     config_stub.val.completion.quick = True
 
     _set_cmd_prompt(status_command_stub, ':open |')
-    completer_obj.on_selection_changed('http://example.com')
-    completer_obj.schedule_completion_update()
-    assert not completion_widget_stub.set_model.called
+
+    url = 'http://example.com'
+    completer_obj._change_completed_part = unittest.mock.Mock()
+    completer_obj.on_selection_changed(url)
+
+    # no immediate (default is false)
+    completer_obj._change_completed_part.assert_called_with(url,      # text
+                                                            ['open'], # before
+                                                            [])       # after
```

Comment:
There should be two spaces before the comments here, but then the line is too long. What about:

```python
    completer_obj._change_completed_part.assert_called_with(
        # text, before, after
        url, ['open'], [])
```

Reply:
I'd rather not check private method calls in a test. What we want to verify here is:

1. Type `:open `
2. See completions
3. Tab to select first completion
4. Completion view does not 'flicker' (close then open again)

The symptom was caused by `set_model` getting called needlessly. The following should test it without needing to mock a private method:

```diff
diff --git a/tests/unit/completion/test_completer.py b/tests/unit/completion/test_completer.py
index 934872f2b..a32241621 100644
--- a/tests/unit/completion/test_completer.py
+++ b/tests/unit/completion/test_completer.py
@@ -303,12 +303,11 @@ def test_quickcomplete_flicker(status_command_stub, completer_obj,
     config_stub.val.completion.quick = True
 
     _set_cmd_prompt(status_command_stub, ':open |')
+    completer_obj.schedule_completion_update()
+    assert completion_widget_stub.set_model.called
+    completion_widget_stub.set_model.reset_mock()
 
-    url = 'http://example.com'
-    completer_obj._change_completed_part = unittest.mock.Mock()
-    completer_obj.on_selection_changed(url)
-
-    # no immediate (default is false)
-    completer_obj._change_completed_part.assert_called_with(url,      # text
-                                                            ['open'], # before
-                                                            [])       # after
+    # selecting a completion should not re-set the model
+    completer_obj.on_selection_changed('http://example.com')
+    completer_obj.schedule_completion_update()
+    assert not completion_widget_stub.set_model.called
```